lib/src/line_intersect.dart

```diff
@@ -0,0 +1,167 @@
+
+import '../helpers.dart';
+
+/**
+ * Takes any LineString or Polygon GeoJSON and returns the intersecting point(s).
+ *
+ * @name lineIntersect
+ * @param {GeoJSON} line1 any LineString or Polygon
+ * @param {GeoJSON} line2 any LineString or Polygon
+ * @param {Object} [options={}] Optional parameters
+ * @param {boolean} [options.removeDuplicates=true] remove duplicate intersections
+ * @param {boolean} [options.ignoreSelfIntersections=false] ignores self-intersections on input features
+ * @returns {FeatureCollection<Point>} point(s) that intersect both
+ * @example
+ * var line1 = turf.lineString([[126, -11], [129, -21]]);
+ * var line2 = turf.lineString([[123, -18], [131, -14]]);
+ * var intersects = turf.lineIntersect(line1, line2);
+ *
+ * //addToMap
+ * var addToMap = [line1, line2, intersects]
+ */
+FeatureCollection<Point>  lineIntersect(
```

Comment:
please check lib/src/intersection. If we have some code-sharing possibility, please use it!

Reply:
Why does lib/src/intersection only handles only two Positions of each line? Should we not consider LineStrings with coordinates.length > 2 as well?
`
  final x1 = line1.coordinates[0][0]!;
  final y1 = line1.coordinates[0][1]!;
  final x2 = line1.coordinates[1][0]!;
  final y2 = line1.coordinates[1][1]!;
  final x3 = line2.coordinates[0][0]!;
  final y3 = line2.coordinates[0][1]!;
  final x4 = line2.coordinates[1][0]!;
  final y4 = line2.coordinates[1][1]!;`

Reply:
Because that is what this function does. It is only doing it for one line segment at a time. Should also be described in the docs.